BadApi/Controllers/Base/BaseApiController.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Web;
+using System.Web.Http;
+
+namespace BadApi.Controllers.Base
+{
+	public class BaseApiController : ApiController
+	{
+		protected IDbConnection GetConnection()
+		{
+			var connection = new SqlConnection("Server=localhost;Database=BadApi;User Id=badapi_user;Password=p@ssw0rd;");
```

Comment:
Consider not hardcoding configuration values in the code itself. Reference application configuration instead.
Also consider using external configuration management system to store and provide configuration values including usernames and passwords at deploy time into the application configuration
Passwords (especially prod) should never be pain text!!